tests/ui/unnecessary_min_max.fixed

```diff
@@ -0,0 +1,56 @@
+#![allow(unused)]
+#![warn(clippy::unnecessary_min_max)]
+fn main() {
+    // Both are Literals
+    let _ = (-6_i32);
+    let _ = 9;
+    let _ = 6;
+    let _ = 9_u32;
+    let _ = 6;
+    let _ = 7_u8;
+
+    // signed MIN
+    let _ = i32::MIN;
+    let _ = 42;
+    let _ = i32::MIN;
+    let _ = 42;
+
+    // signed MAX
+    let _ = 42;
+    let _ = i32::MAX;
+    let _ = 42;
+    let _ = i32::MAX;
+
+    // unsigned MAX
+    let _ = 42;
+    let _ = u32::MAX;
+    let _ = 42;
+    let _ = u32::MAX;
+
+    // unsigned MIN
+    let _ = u32::MIN;
+    let _ = 42;
+    let _ = u32::MIN;
+    let _ = 42;
+
+    // unsigned with zero
+    let _ = 0;
+    let _ = 7_u8;
+    let _ = 0_u32;
+    let _ = 7;
+
+    // The below cases shouldn't be lint
+    let mut min = u32::MAX;
+    for _ in 0..1000 {
+        min = min.min(random_u32());
+    }
+
+    let _ = (u32::MIN as isize).min(42);
```

Comment:
An implementation detail of `constant` but I don't think it handles `as` casts, you'd need to either put these in a `const` or inline the value as e.g. `2_147_483_647isize`